src/passes/warnSupportedFeatures.ts

```diff
@@ -0,0 +1,70 @@
+import fs from 'fs';
+import {
+  AddressType,
+  ASTNode,
+  ExternalReferenceType,
+  FunctionCall,
+  FunctionCallKind,
+  parseSourceLocation,
+} from 'solc-typed-ast';
+import { AST } from '../ast/ast';
+import { ASTMapper } from '../ast/mapper';
+import { warning } from '../utils/formatting';
+import { safeGetNodeType } from '../utils/nodeTypeProcessing';
+import { getSourceFromLocations } from '../utils/utils';
+
+export class WarnSupportedFeatures extends ASTMapper {
+  public addressesToAbiEncode: ASTNode[] = [];
+
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    if (
+      node.kind === FunctionCallKind.FunctionCall &&
+      node.vFunctionCallType === ExternalReferenceType.Builtin &&
+      ['encodePacked'].includes(node.vFunctionName)
+    ) {
+      node.vArguments
+        .filter((arg) => safeGetNodeType(arg, ast.compilerVersion) instanceof AddressType)
+        .forEach((arg) => this.addressesToAbiEncode.push(arg));
+    }
+
+    return this.visitExpression(node, ast);
```

Comment:
This is fine, but `commonVisit` is usually used instead. Plus, `visitExpression` does `commonVisit` on the node, so it might as well be used directly